### PR TITLE
Add quad8 extrusion, fixed mixed order extrusion

### LINKED ISSUE #27661: QUAD8 extrusion into HEX20
### Motivation

Reed and I are exploring options for meshing cells.
We noticed quad8 wasnt available for extrusion

and the error message is a cryptic "libmesh_unimplemented" which could be more friendly. I would be nice to know what is not implemented for example without pulling out a debugger.

### Design

Add one more element type for extrusion
This is actually all of them then?

### Impact

A little more capability
Better error

### PR DESCRIPTION
**Pull request by GiudGiud**
closes #27661

### CODE REVIEW

File: framework/src/meshgenerators/AdvancedExtruderGenerator.C [hunk 1]
```diff
@@ -742,6 +753,74 @@ AdvancedExtruderGenerator::generate()
...
+            if (elem->neighbor_ptr(0) == remote_elem)
+              new_elem->set_neighbor(1, const_cast<RemoteElem *>(remote_elem));
+            if (elem->neighbor_ptr(1) == remote_elem)
+              new_elem->set_neighbor(2, const_cast<RemoteElem *>(remote_elem));
+            if (elem->neighbor_ptr(2) == remote_elem)
+              new_elem->set_neighbor(3, const_cast<RemoteElem *>(remote_elem));
+            if (elem->neighbor_ptr(3) == remote_elem)
+              new_elem->set_neighbor(4, const_cast<RemoteElem *>(remote_elem));
+
+            if (new_elem->volume() < 0.0)
```
Comment (roystgnr) [inline framework/src/meshgenerators/AdvancedExtruderGenerator.C:809]: Could this whole block be replaced by a new_elem->orient() call?

File: framework/src/meshgenerators/AdvancedExtruderGenerator.C [hunk 2]
```diff
@@ -993,6 +1072,10 @@ AdvancedExtruderGenerator::generate()
                                                             input_nodeset_map.end());
 
   mesh->set_isnt_prepared();
+  // Creating the layered meshes creates a lot of leftover nodes, notably in the boundary_info,
+  // which will crash both paraview and trigger exodiff. Best to be safe.
+  if (extruding_quad_eights)
+    mesh->prepare_for_use();
```
Comment (roystgnr) [inline framework/src/meshgenerators/AdvancedExtruderGenerator.C:1078]: If the only thing we need here is orphaned node removal, can we get away with `mesh->remove_orphaned_nodes()`?  Or maybe with `mesh->renumber_nodes_and_elements()`, if contiguous numbering is preferred and renumbering is allowed?

**Review by roystgnr — APPROVED** [review]

File: framework/src/meshgenerators/AdvancedExtruderGenerator.C (hunk 1 above)
Comment (GiudGiud) [reply framework/src/meshgenerators/AdvancedExtruderGenerator.C:809]: I like that the way we do it here is very explicitly flipping top and bottom. Orient could be flipping left and right

**Comment by GiudGiud** [comment]
wow force pushing back to same commit gets me my test results back
civet so smart

**Comment by GiudGiud** [comment]
Thanks for the review!

**Pull request merged**